Download and install the TensorFlow 2.0 Beta package. Import TensorFlow into your program

In [2]:
from __future__ import absolute_import, division, print_function, unicode_literals

# Install TensorFlow
try:
  # %tensorflow_version only exists in Colab.
  %tensorflow_version 2.x
except Exception:
  pass

import tensorflow as tf

TensorFlow 2.x selected.


Load and prepare the [MNIST dataset](http://yann.lecun.com/exdb/mnist/). Convert the samples from integers to floating-point numbers:

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11493376/11490434 [==============================] - 0s 0us/step


Build the `tf.keras.Sequential` model by stacking layers. Choose an optimizer and loss function for training:

In [0]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

![alt text](https://)Train and evaluate the model:

In [6]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 5s 89us/sample - loss: 0.2964 - accuracy: 0.9125
Epoch 2/5
60000/60000 [==============================] - 5s 80us/sample - loss: 0.1442 - accuracy: 0.9579
Epoch 3/5
60000/60000 [==============================] - 5s 81us/sample - loss: 0.1073 - accuracy: 0.9677
Epoch 4/5
60000/60000 [==============================] - 5s 80us/sample - loss: 0.0853 - accuracy: 0.9737
Epoch 5/5
10000/10000 [==============================] - 1s 50us/sample - loss: 0.0816 - accuracy: 0.9751


[0.08155284733781591, 0.9751]

The image classifier is now trained to ~98% accuracy on this dataset.

---

